test/renate/benchmark/test_scenarios.py

```diff
@@ -197,3 +198,27 @@ def test_iid_scenario():
                 assert x not in counter
                 counter[x] = 1
         assert len(scenario.test_data()) == 3
+
+
+def test_soft_sorting_scenario():
+    """Tests that the soft-sorting scenario creates a non-overlapping split."""
+    data_module = DummyTorchVisionDataModule(val_size=0.3)
+    counter = {}
+    for i in range(3):
+        scenario = SoftSortingScenario(
+            data_module=data_module,
+            num_tasks=3,
+            feature_idx=0,
+            exponent=10,
+            chunk_id=i,
+            seed=data_module._seed,
+        )
+        scenario.prepare_data()
+        scenario.setup()
+        for stage in ["train", "val"]:
+            scenario_data = getattr(scenario, f"{stage}_data")()
+            for j in range(len(scenario_data)):
+                x, y = scenario_data[j]
+                assert x not in counter
+                counter[x] = 1
```

Comment:
This only checks for the absence of duplicates.
Can we check the sorting as well? maybe using an extremely large exponent?

Reply:
I added a check for a decreasing _average_ value of the relevant feature between chunks.